### PR TITLE
Enable Create/PasswordChange/SignIn date tracking + Store versioning

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/Identity/issues/650

@ajcvickers we can review/discuss this in person, this actually is a good cross cutting PR that highlights how identity is designed, and how typically we add features.

Things to consider:
- Is adding a single "UserActivityStore" better than adding 3 stores (one per property), the amount of stores we have is getting a bit out of hand so my default is to try to consolidate.
- RefreshSignIn does not update the last sign in time (since this is used when cookies are renewed automatically)
- This might be a good opportunity to add some basic automated schema migration test coverage, since this introduces our first schema change from 1.0 (3 new DateTimeOffset?properties).  Can someone point me to any ef tests that might be relevant?

cc @divega @blowdart

### CODE REVIEW

File: test/Microsoft.AspNetCore.Identity.Test/SignInManagerTest.cs [hunk 1]
```diff
@@ -543,23 +562,39 @@ public override Task<bool> ValidateAsync(string purpose, string token, UserManag
...
-        [InlineData(true, true, true, false)]
-        [InlineData(true, true, false, false)]
-        [InlineData(true, false, true, false)]
-        [InlineData(true, false, false, false)]
-        [InlineData(false, true, true, false)]
-        [InlineData(false, true, false, false)]
-        [InlineData(false, false, true, false)]
-        [InlineData(false, false, false, false)]
-        public async Task CanTwoFactorSignIn(bool isPersistent, bool supportsLockout, bool externalLogin, bool rememberClient)
+        [InlineData(true, true, true, true, true)]
```
Comment (HaoK) [inline test/Microsoft.AspNetCore.Identity.Test/SignInManagerTest.cs]: I'm not sure how I feel about this block of 32 permutations of true/false, but I guess its a good way to test every permutation :/  But I am not looking forward to any doubling in the future...

File: test/Microsoft.AspNetCore.Identity.Test/UserManagerTest.cs [hunk 2]
```diff
@@ -694,146 +705,6 @@ public void UserManagerWillUseTokenProviderInstance()
         }
 
         [Fact]
-        public void TOTPTest()
```
Comment (HaoK) [inline test/Microsoft.AspNetCore.Identity.Test/UserManagerTest.cs]: Unrelated test cleanup

**Comment by HaoK** [comment]
This is also a building block for the adding password expiration support, which will be in its own PR: https://github.com/aspnet/Identity/issues/99

**Comment by HaoK** [comment]
Updated this PR to version the store support, boiled down to:
- `IdentityStoreOptions.Version` in `Microsoft.Extensions.Identity.Stores` can be used by stores to version themselves.
- The default implementations of the UserStores now have a `UserStoreBaseV1` and `UserStoreBaseV2` with the V2 extending V1 and additionally implementing the new `IUserActitvityStore`.
- `AddEntityFrameworkStores[V1|V2]()` add the appropriate `UserStoreV1/V2`
- Still need a way to access `IOptions<IdentityStoreOptions>` from a DbContext @ajcvickers (I hardcoded things in the tests for now)
- This should also enable us to directly support a new `IdentityClientApplication` table to eliminate the need for a separate Identity.Service.EntityFrameworkCore package if we add it to the 2.0 schema changes, and add a new `ClientApplicationManager`/`IClientApplicationStore` to `Microsoft.Extensions.Identity.Core` @javiercn @danroth27

**Comment by HaoK** [comment]
@ajcvickers this is the PR with the updated store versioning support

**Comment by HaoK** [comment]
This was done via be8232d3f9c7a312314ab42dfe37047ec3b16665

**Comment by vladimirlapacek** [comment]
I am not sure, how to read this. Has the change ever been merged back to `dev`? Or has it been abandoned?

**Comment by HaoK** [comment]
This was reverted and is waiting for the proper time to bring back
